optaplanner-core/src/benchmark/java/org/optaplanner/core/impl/heuristic/move/AbstractPlannerMoveMicroBenchmark.java

```diff
@@ -0,0 +1,64 @@
+package org.optaplanner.core.impl.heuristic.move;
+
+import java.util.concurrent.TimeUnit;
+
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+import org.optaplanner.core.impl.score.director.ScoreDirector;
+
+@State(Scope.Thread)
+@BenchmarkMode(Mode.AverageTime)
+@Warmup(iterations = 5, time = 3)
+@Measurement(iterations = 5, time = 5)
+@Fork(value = 5)
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+public abstract class AbstractPlannerMoveMicroBenchmark<Solution> {
```

Comment:
@Fonifan Running it locally I see following results:
Benchmark                            Mode  Cnt     Score      Error  Units
ChangeMoveBenchmark.benchmarkDoMove  avgt   25  4043.934 ± 1203.210  ns/op
ChangeMoveBenchmark.benchmarkRebase  avgt   25   451.929 ±   96.274  ns/op

The error looks scary; 4043.934 ± 1203.210 is not much reliable result. What numbers do you have locally?

Reply:
I have the same settings on my machine, but it seemed to be more stable (situation like this would occur once in 10 runs, so I thought it will be more stable on perf-machine and thus these settings will be a nice compromise between time and stability). 

Increasing number of warmup iterations to 10 seems to stabilize the benchmark, but it takes around 10 minutes for a single run. Are these optimal numbers?

Reply:
Depends - does it improve if you increase warmup iterations? Or warmup time? Do we need so many forks? Without having reliability of results below 5%, there is no point in having such a benchmark. On laptop, the reliability may be worse, let's say between 5 and 10%, but anything more makes the benchmark unusable for a developer who would like to run it locally.

Reply:
After experimenting with settings, I've managed to find a good compromise -> 3 forks, 10 warmup iterations.
Results from 3 runs:
* ChangeMoveBenchmark.benchmarkDoMove  avgt   15  2591.209 ± 85.654  ns/op
* ChangeMoveBenchmark.benchmarkDoMove  avgt   15  2573.370 ± 85.696  ns/op
* ChangeMoveBenchmark.benchmarkDoMove  avgt   15  2534.195 ± 196.403  ns/op